test_unstructured/partition/test_msg.py

```diff
@@ -156,102 +132,49 @@ def test_partition_msg_from_file_exclude_metadata():
         assert elements[i].metadata.to_dict() == {}
 
 
-def test_partition_msg_can_process_attachments(tmp_path: pathlib.Path):
-    file_path = example_doc_path("fake-email-attachment.msg")
-    tmp_dir_path = str(tmp_path)
-    extract_msg_attachment_info(filename=file_path, output_dir=tmp_dir_path)
-    attachment_filename = str(tmp_path / str(ATTACH_EXPECTED_OUTPUT[0]["filename"]))
-
-    mocked_last_modification_date = "2029-07-05T09:24:28"
-
-    attachment_elements = partition_text(
-        filename=attachment_filename,
-        metadata_filename=attachment_filename,
-        metadata_last_modified=mocked_last_modification_date,
-    )
-    expected_metadata = attachment_elements[0].metadata
-    expected_metadata.file_directory = None
-    expected_metadata.attached_to_filename = file_path
-
-    elements = partition_msg(
-        filename=file_path,
-        attachment_partitioner=partition_text,
-        process_attachments=True,
-        metadata_last_modified=mocked_last_modification_date,
-    )
-
-    # This test does not need to validate if hierarchy is working
-    # Patch to nullify parent_id
-    expected_metadata.parent_id = None
-    elements[-1].metadata.parent_id = None
-
-    assert elements[0].text.startswith("Hello!")
-    for element in elements[:-1]:
-        assert element.metadata.filename == "fake-email-attachment.msg"
-        assert element.metadata.subject == "Fake email with attachment"
-    assert elements[-1].text == "Hey this is a fake attachment!"
-    assert elements[-1].metadata == expected_metadata
-
-
-def test_partition_msg_can_process_min_max_wtih_attachments(tmp_path: pathlib.Path):
-    file_path = example_doc_path("fake-email-attachment.msg")
-    tmp_dir_path = str(tmp_path)
-    extract_msg_attachment_info(filename=file_path, output_dir=tmp_dir_path)
-    attachment_filename = str(tmp_path / str(ATTACH_EXPECTED_OUTPUT[0]["filename"]))
-
-    attachment_elements = partition_text(
-        filename=attachment_filename,
-        metadata_filename=attachment_filename,
-        min_partition=6,
-        max_partition=12,
-    )
-
+def test_partition_msg_can_process_attachments():
```

Comment:
Do you know how this test was passing before if attachment processing was broken?

Reply:
Hmm, good question.

The test file here: `example-docs/fake-email-attachment.msg`, has one plain-text attachment: `fake-attachment.txt`.

My hypothesis would be it would have to do with `msg_parser` improperly applying some encoding to the attachment bytes. The encoding for the HTML body for this message is UTF-8 and if the attachment was "decoded" with that codec it would definitely corrupt the attachment but leave the "PK" prefix intact, which is what we observed in #3006. In the MSG format, attachment bytes are opaque `bytes` and should be detached intact.

I didn't try to unravel the spaghetti to pinpoint the cause of corruption, figuring it wasn't going to be as simple as just submitting a fix PR to `msg_parser` :)

Reply:
Yeah, was mainly confused why it wasn't failing before since that was the test file that was causing the issue. Don't think we need to unravel why that was happening in `msg_parser`, but it would be nice to have a test for that file that fails on `main` but passes on this feature branch to verify we've fixed #3006. Unless you think that's captured in one of the other tests in here.

Reply:
The test file that was failing was `fake-email-multiple-attachments.msg`. That one has three attachments, one of which is a PPTX and that's the one the failed. This `fake-email-attachment.msg` just has one plain-text attachment.

I can change this test to use the one with the PPTX instead, sounds like a great idea to me.

I'd have to make a different PR to add an XFAIL version if we actually wanted it on main, but if you're satisfied with me trying it on main and seeing it fail there then we can just pop it in here.

Reply:
k, I went ahead and did that here: https://github.com/Unstructured-IO/unstructured/pull/3142/files?show-viewed-files=true&file-filters%5B%5D=#diff-28bb41388f1944285315370fad3dad611289870db16e7962e8c94565bdd534a1R135

Reply:
Awesome, thanks!